e2e/puppeteer-custom-environment.ts

```diff
@@ -16,85 +26,72 @@ class PuppeteerCustomEnvironment extends PuppeteerEnvironment {
-  // 20201103_084435
-  localDateTimeString() {
-    const localTime = new Date();
-    const fakeUtcTime = new Date(localTime.getTime() - localTime.getTimezoneOffset() * 60000);
-    return fakeUtcTime.toISOString().replace(/[-:]/g, '').slice(0, 15);
+  getVmContext() {
+    return super.getVmContext();
+  }
+
+  getNames(parent) {
+    if (!parent || parent.name === 'ROOT_DESCRIBE_BLOCK') {
+      return [];
+    }
+    const parentName = this.getNames(parent.parent);
+    return [...parentName, parent.name];
   }
 
   // jest-circus: https://github.com/facebook/jest/blob/master/packages/jest-circus/README.md#overview
-  // Take a screenshot right after failure
   async handleTestEvent(event, state) {
-    switch (event.name) {
-      case 'test_start':
-        // https://stackoverflow.com/questions/51250006/jest-stop-test-suite-after-first-fail
-        // When one test fails, next tests in the same describe block will be skipped.
-        // If there are more describe blocks in the same test file, they will not be skipped.
-        if (this.failedTestSuites[event.test.parent.name]) {
-          event.test.mode = 'skip';
-        }
-        break;
-      case 'test_fn_failure':
-      case 'hook_failure':
-        {
-          let describeName;
-          try {
-            describeName = state.currentlyRunningTest.parent.name;
-          } catch (err) {
-            describeName = event.test.parent.name;
-          }
-          this.failedTestSuites[describeName] = true;
-          const runningTest = state.currentlyRunningTest;
-          let testName;
-          if (runningTest != null) {
-            testName = runningTest.name.replace(/\W/g, '-');
-          } else {
-            testName = event.test.name.replace(/\W/g, '-');
-          }
-          await fs.ensureDir(this.screenshotDir);
-          await fs.ensureDir(this.htmlDir);
-          const screenshotFile = `${this.screenshotDir}/${testName}.png`;
-          await this.takeScreenshot(screenshotFile);
-          const htmlFile = `${this.htmlDir}/${testName}.html`;
-          await this.savePageToFile(htmlFile);
-        }
-        break;
-      default:
-        break;
+    const { name } = event;
+    if (['test_start', 'test_fn_start'].includes(name)) {
+      this.global.__TEST_NAMES__ = this.getNames(event.test);
+      // https://stackoverflow.com/questions/51250006/jest-stop-test-suite-after-first-fail
+      // When one test fails, next tests in the same describe block will be skipped.
+      // If there are more describe blocks in the same test file, they will not be skipped.
+      if (this.failedTestSuites[this.global.__TEST_NAMES__[0]]) {
+        event.test.mode = 'skip';
+      }
+    }
+    if (['hook_failure', 'test_fn_failure'].includes(name)) {
+      const describeBlockName = this.global.__TEST_NAMES__[0];
+      this.failedTestSuites[describeBlockName] = true;
+      // TODO REMOVE WHEN DELETING JEST-PUPPETEER
```

Comment:
Same as above comment. Also please rewrite without capslock: https://github.com/all-of-us/workbench/wiki/UI-Style-Guide#code-comments

Reply:
Done. I updated [RW-6967](https://precisionmedicineinitiative.atlassian.net/browse/RW-6967) with steps (reminders).